.github/workflows/create-release-main.yml

```diff
@@ -0,0 +1,71 @@
+name: Create new preview release (main)
+
+on:
+  # run when pushed to main is published,
+  # which creates a new tag
+  push:
+    branches:
+      - main
+
+env:
+  ARTIFACT_VERSION: experimental
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # required to access tags
+          submodules: "true"
+
+      - name: Build & run devcontainer image
+        # this always builds a new image from scratch rather than from the build-devcontainer-image workflow output
+        # so that we pick up the latest versions of everything
+        # NB: if you update this also update live-validation.yml
+        id: devcontainer
+        run: |
+          docker build --tag devcontainer:latest .devcontainer
+          mkdir -p $HOME/.docker # in case it doesn't exist
+          container_id=$(docker create -w /workspace -v $GITHUB_WORKSPACE:/workspace -v /var/run/docker.sock:/var/run/docker.sock devcontainer:latest)
+          docker start "$container_id"
+          echo "container_id=$container_id" >> $GITHUB_ENV
+
+      - name: Build required release files
+        run: |
+          container_id=${{env.container_id}}
+          docker exec "$container_id" task VERSION=${{ env.ARTIFACT_VERSION }} LATEST_VERSION_TAG=${{ env.ARTIFACT_VERSION }} make-release-artifacts
+
+      - name: Upload release assets
+        uses: svenstaro/upload-release-action@7319e4733ec7a184d739a6f412c40ffc339b69c7 # this is v2.5.0, but pinned
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          tag: refs/tags/${{ env.ARTIFACT_VERSION }}
+          file: "v2/out/release/*"
+          file_glob: true
+          prerelease: true
+          overwrite: true
+          release_name: "Experimental"
+          body: |
+            This is the "latest" releasewhich contains the most recent commits from the main branch. (commit: ${{ github.sha }}).
+
+            **This release might not be stable.**
+
+            It is only intended for developers wishing to try out the latest feature, some of which may not be fully implemented.
+
+      - name: Docker login
+        run: |
+          container_id=${{env.container_id}}
+          docker exec -e AZURE_CLIENT_ID -e AZURE_CLIENT_SECRET -e DOCKER_REGISTRY "$container_id" task VERSION=${{ env.ARTIFACT_VERSION }} LATEST_VERSION_TAG=${{ env.ARTIFACT_VERSION }} docker-login
```

Comment:
Docker login would only require `AZURE_CLIENT_ID` and `AZURE_CLIENT_SECRET` here.

Reply:
That is what i thought, but it tries to resolve the version anyway for some reason.

Example:
- Commit: https://github.com/tomkerkhove/azure-service-operator/commit/6d5345a1debab58f736c9e4f19c8857545583bd2
- Failure because Git tags are not applicable here: https://github.com/tomkerkhove/azure-service-operator/actions/runs/7503829500/job/20429481194